WordPress/gutenberg
# ToolsPanel: Refine component behaviour

Fixes: https://github.com/WordPress/gutenberg/issues/34257

Depends on:
- https://github.com/WordPress/gutenberg/pull/34632

## Description

- Updates the `ToolsPanel` to improve its UX following the [suggestions outlined](https://github.com/WordPress/gutenberg/issues/34257#issuecomment-908554545) on the corresponding issue.
- Adds min-width to `ToolsPanel` dropdown menu
- Adds aria-labels to menu items

## How has this been tested?

Manually:

* Tested using the group block with padding and margin support
* Tested using changes from [34065](https://github.com/WordPress/gutenberg/pull/34065) and its injected min-height control for the cover block
* Manually updated storybook story to test various configurations

#### Test instructions
1. For testing, update the Group block's config to opt into padding and margin support, enabling one as a default control. 
```
		"spacing": {
			"padding": true,
			"margin": true,
			"__experimentalDefaultControls": {
				"padding": true
			}
		},
```
2. Depending on your theme, you'll likely also need to enable the margin support UI by setting the `settings.spacing.customMargin` property to `true` in your theme.json file.
3. Add a group block in the post editor and select it
4. Explore the behaviour of the ToolsPanel via the block's "Dimensions" panel.
5. Fire up native and confirm that the `Stepper` component displays its check icon correctly. Any `UnitControl` on native will show this `Stepper` component, one easy block to test with on native is the Cover block.

## Screenshots 

#### Cover Block with cherry-picked commit from #34065
https://user-images.githubusercontent.com/60436221/132173157-a41ff448-2b83-43bb-ad54-877839013e25.mp4

#### Storybook
 https://user-images.githubusercontent.com/60436221/132362438-1a9ac63d-841d-4e62-ada9-d80b06b51e1e.mp4

## Types of changes
Enhancement

## Checklist:
